### ISSUE TITLE
When I save file Chinese will be converted into ���  ，Does anyone know why?

### ISSUE THREAD

**Issue by chenenze** [Microsoft/vscode-go#1469]
![image](https://user-images.githubusercontent.com/9709847/35025799-86bf181c-fb82-11e7-8f01-63b8476f5351.png)

The problem disappeared when I uninstall this plugin

**Comment by ramya-rao-a** [comment]
On save, the only extra edits being performed are the formatting edits. Install the extension and try the below

- Add `"go.formatTool": "gofmt"` to your settings and reload VS Code. Do you still see the issue?
- If you still see the issue, then add the below to your settings and try again
```
"[go]": {
    "editor.formatOnSave": false
   }
```

**Comment by chenenze** [comment]
Thank you for your help! It's working

**Comment by chenenze** [comment]
This problem reappeared when I was working for some time
This is my setting：
{
    "team.showWelcomeMessage": false,
    "go.formatTool": "gofmt",
    "go.formatOnSave": false,
    "files.autoSave": "afterDelay",
    "files.autoSaveDelay": 2000,
    "workbench.iconTheme": "vscode-icons",
}

**Comment by ramya-rao-a** [comment]
Can you get a consistent repro of the issue or does it happen randomly? If you can repro it consistently, then can you share a small sample code where you see the issue?

**Comment by chenenze** [comment]
It's not randomly, When I click command+s save the file, This situation is repeated，when I delete all the functions except this function，this problem has disappeared.

I try to write a testcase like this：
```go
package main

func test() {
    // 仅在用户点击【通过】时才确认
}
```
But did not reproduce。

**Comment by ramya-rao-a** [comment]
@chenenze I cannot help if I don't have a file that I can use to see the issue :(

**Comment by xfstart07** [comment]
I have the same problem。
![image](https://user-images.githubusercontent.com/950787/35181268-84cea378-fdf9-11e7-90a3-ac6d910dff2f.png)

**Comment by ramya-rao-a** [comment]
@xfstart07 Please share a sample code where you can reproduce the issue consistently

**Comment by ramya-rao-a** [comment]
Thanks @molon, I can repro the issue, but am not sure what the root cause is.

Until I get around to fixing this, the workaround is to either disable the formatOnSave feature by adding  the below to your settings
```
"[go]": {
 "editor.formatOnSave": false
}
```

OR you can downgrade to 0.6.69 version of the Go extension by
- Disable auto-updating of extensions
- Downloading the vsix file from https://github.com/Microsoft/vscode-go/releases/tag/0.6.69
- Run `code --install-extension Go-0.6.69.vsix` and restart VS Code

**Comment by ramya-rao-a** [comment]
Alright, I found the problem!

Recently, I moved from using the [execFile](https://nodejs.org/api/child_process.html#child_process_child_process_execfile_file_args_options_callback) to [spawn](https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options) when creating the child process that runs the formatting tool.

When using execFile there is default utf-8 encoding that happens. When using spawn, the streams aren't encoded by default and we need to [set the right encoding](https://nodejs.org/api/stream.html#stream_readable_setencoding_encoding) to ensure that the multi-byte content in stdout gets handled properly. 

I'll release an update with the fix in the next week or two. Until then, you can get the fix by doing the below:
- Download https://github.com/Microsoft/vscode-go/blob/master/Go-latest.vsix 
- run `code --install-extension Go-latest.vsix`
- If the above fails with `Error: end of central directory record signature not found`, then clone this repo and use the `Go-latest.vsix` file from the cloned repo
- Reload VS Code

**Comment by beta** [comment]
Thank you Ramya for your work! Another workaround before the issue is fixed is to run the formatter automatically with extensions that allow you to run commands on file save. These extensions include [Run on Save](https://marketplace.visualstudio.com/items?itemName=emeraldwalk.RunOnSave) and [Trigger Task on Save](https://marketplace.visualstudio.com/items?itemName=Gruntfuggly.triggertaskonsave). I am using Run on Save with the following configuration.

```json
"emeraldwalk.runonsave": {
    "commands": [{
        "match": "\\.go$",
        "cmd": "goreturns -w ${file}"
    }]
}
```

Of course you should disable the built-in auto formatter with

```json
"[go]": {
    "editor.formatOnSave": false
}
```

**Comment by ramya-rao-a** [comment]
Just released an update (0.6.74) that has the fix for this issue. Thanks for your patience everyone!
